reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConnect.java

```diff
@@ -242,13 +242,13 @@ public void subscribe(CoreSubscriber<? super Connection> actual) {
 				else {
 					if (_config.sslProvider != null) {
 						_config = new HttpClientConfig(config);
-						if (_config.checkProtocol(HttpClientConfig.h2) && _config.protocols.length > 1) {
-							removeIncompatibleProtocol(_config, HttpProtocol.H2);
-						}
 						_config.sslProvider = null;
 					}
 
-					if (_config.checkProtocol(HttpClientConfig.h2)) {
+					if (_config.checkProtocol(HttpClientConfig.h2) && _config.protocols.length > 1) {
+						removeIncompatibleProtocol(_config, HttpProtocol.H2);
+					}
+					else if (_config.checkProtocol(HttpClientConfig.h2)) {
```

Comment:
What do you think about checking once for `_config.checkProtocol(HttpClientConfig.h2)`?

Reply:
Sounds good.